core/frontend/src/test/tile/GltfReader.test.ts

```diff
@@ -437,4 +439,635 @@ describe("GltfReader", () => {
+              propertyCount0++;
+            } else if(key === "INT8_VALUES") {
+              expect(value === expectedValuesSigned[entryCount]).to.be.true;
+              propertyCount0++;
+            } else if(key === "INT16_VALUES") {
+              expect(value === expectedValuesSigned[entryCount]).to.be.true;
+              propertyCount0++;
+            } else if(key === "INT32_VALUES") {
+              expect(value === expectedValuesSigned[entryCount]).to.be.true;
+              propertyCount0++;
+            } else if(key === "INT64_VALUES") {
+              expect(value === expectedValuesBigSigned[entryCount]).to.be.true;
+              propertyCount0++;
+            } else if(key === "FLOAT32_VALUES") {
+              expect(value === expectedValuesUnsigned[entryCount]).to.be.true;
+              propertyCount0++;
+            } else if(key === "FLOAT64_VALUES") {
+              expect(value === expectedValuesUnsigned[entryCount]).to.be.true;
+              propertyCount0++;
+            } else if(key === "STRING_VALUES") {
+              expect(value === expectedValuesString[entryCount]).to.be.true;
+              propertyCount0++;
+            }
+          }
+          expect(propertyCount0 === 11).to.be.true;
+        }
+
+        // Expect empty property set for null feature id = 3
+        if(instanceFeatures[entryCount] === 3){
+          expect(JSON.stringify(entry.properties.propertySet1) === JSON.stringify({})).to.be.true;
+        } else {
+          let propertyCount1 = 0;
+          for(const [key, value] of Object.entries(entry.properties.propertySet1)){
+            const array = value as any[];
+            if(key === "UINT8_VEC2_VALUES") {
+              expect(compareArrays(array, expectedValuesVec2[instanceFeatures[entryCount]] )).to.be.true;
+              propertyCount1++;
+            } else if(key === "UINT8_VEC3_VALUES") {
+              expect(compareArrays(array, expectedValuesVec3[instanceFeatures[entryCount]] )).to.be.true;
+              propertyCount1++;
+            } else if(key === "UINT8_VEC4_VALUES") {
+              expect(compareArrays(array, expectedValuesVec4[instanceFeatures[entryCount]] )).to.be.true;
+              propertyCount1++;
+            } else if(key === "UINT8_MAT2_VALUES") {
+              expect(compareArrays(array, expectedValuesVec4[instanceFeatures[entryCount]] )).to.be.true;
+              propertyCount1++;
+            } else if(key === "UINT8_MAT3_VALUES") {
+              expect(compareArrays(array, expectedValuesMat3[instanceFeatures[entryCount]] )).to.be.true;
+              propertyCount1++;
+            }else if(key === "UINT8_MAT4_VALUES") {
+              expect(compareArrays(array, expectedValuesMat4[instanceFeatures[entryCount]] )).to.be.true;
+              propertyCount1++;
+            }
+          }
+          expect(propertyCount1 === 6).to.be.true;
+        }
+
+        entryCount ++;
+      }
+      expect(entryCount === 4).to.be.true;
```

Comment:
Don't make any changes but in future prefer `expect(entryCount).to.equal(4)` so people don't have to start a debugger or edit the test to find out what `entryCount` actually is when the test fails.

Reply:
Thanks, I knew that there was a reason why this stuff existed, but I could not figure why, now I know 👍